packages/core/realtime-js/src/RealtimeChannel.ts

```
@@ -469,28 +464,59 @@ export default class RealtimeChannel {
+      }
+
+      // Joining while an auth call is still in flight sends the join without an
+      // access token, so the server falls back to the anon apikey and RLS filters
+      // everything out while the channel still reports SUBSCRIBED. Only defer when
+      // there is something to wait for, so the common path stays synchronous.
+      if (this.socket._hasPendingAuth()) {
+        // The caller can leave before the deferred join runs (a logout, a component
+        // unmounting). phoenix's leave() puts a never-joined channel straight back
+        // into `closed`, so the state alone cannot tell the two apart.
+        const sendJoinUnlessLeft = () => {
+          if (!this._leaveRequested) sendJoin()
```

ESLint

[error] Expected { after 'if' condition.

(curly)